### PR TITLE
Add StatefulSet Resource

### PR DESCRIPTION
**Pull request by sarahbx**
##### Short description:
Add StatefulSet Resource

### CODE REVIEW

**Comment by sarahbx** [comment]
!verified

File: ocp_resources/stateful_set.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from ocp_resources.utils import TimeoutSampler
+
+
+LOGGER = get_logger(name=__name__)
+
+
+class StatefulSet(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.APPS
+
+    def scale_replicas(self, replica_count):
```
Comment (myakove) [inline ocp_resources/stateful_set.py:14]: We have this code in other places, I think we can move it to resource.py and reuse it.

File: ocp_resources/stateful_set.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        Returns:
+            StatefulSet is updated successfully
+        """
+        super().to_dict()
+        self.res.update({"spec": {"replicas": replica_count}})
+
+        LOGGER.info(f"Set StatefulSet replicas: {replica_count}")
+        return self.update(resource_dict=self.res)
+
+    def wait_for_replicas(self, deployed=True, timeout=TIMEOUT_4MINUTES):
```
Comment (myakove) [inline ocp_resources/stateful_set.py:28]: We have this code in other places, I think we can move it to resource.py and reuse it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/stateful_set.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+from ocp_resources.constants import PROTOCOL_ERROR_EXCEPTION_DICT, TIMEOUT_4MINUTES
+from ocp_resources.logger import get_logger
+from ocp_resources.resource import NamespacedResource
+from ocp_resources.utils import TimeoutSampler
+
+
+LOGGER = get_logger(name=__name__)
+
+
+class StatefulSet(NamespacedResource):
```
Comment (myakove) [inline ocp_resources/stateful_set.py:11]: Missing link to resource doc

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/stateful_set.py (hunk 2 above)
Comment (sarahbx) [reply ocp_resources/stateful_set.py:28]: This one is slightly different than in `Deployment`
adds `current_replicas`

**Review by rnetser — APPROVED** [review]

File: ocp_resources/stateful_set.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    api_group = NamespacedResource.ApiGroup.APPS
+
+    def __init__(
+        self,
+        name,
+        namespace,
+        service_name,
+        pod_selector,
+        pod_template,
+        replicas,
```
Comment (rnetser) [inline ocp_resources/stateful_set.py:26]: `yaml_file` is missing; and these should be set to `None` to avoid failure

**Review by rnetser — CHANGES_REQUESTED** [review]

**Pull request closed without merging**